### PR TITLE
feat(AccessAPI): add support for JSON patches in access sdk [EXT-3041]

### PR DESCRIPTION
**Pull request by kdamball**
# Purpose of PR

Pointing to master now to release for v3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "contentful-ui-extensions-sdk",
   "description": "SDK to develop custom UI Extension for the Contentful Web App",
-  "version": "3.39.2",
+  "version": "3.39.3",
```
Comment (shikaan) [inline package.json:4]: I think we can let the CI do the bump here

**Review by shikaan — APPROVED** [review]
LGTM we might need to revert package-lock and package.json

File: lib/types.ts [hunk 2]
```diff
@@ -693,6 +698,11 @@ export interface AccessAPI {
     entity: 'ContentType' | ContentType | 'Asset' | 'Entry' | T
   ): Promise<boolean>
   can<T = Object>(action: ArchiveableAction, entity: 'Asset' | 'Entry' | T): Promise<boolean>
+  can<T = Object>(
+    action: 'patch',
+    entity: 'Asset' | 'Entry' | T,
```
Comment (andipaetzold) [inline lib/types.ts:703]: Not quite sure about this generic. Shouldn't it just be 
```suggestion
    entity: 'Asset' | 'Entry' | string,
```
and removing the generic completely?

Or is this just to be consistent and we fix this with v4?

File: lib/types.ts [hunk 3]
```diff
@@ -693,6 +698,11 @@ export interface AccessAPI {
...
+    patch?: JSONPatchItem[]
```
Comment (andipaetzold) [inline lib/types.ts:704]: When can `patch` be `undefined`? Doesn't that basically say "check whether you can keep the entity as it is"

File: lib/types.ts (hunk 2 above)
Comment (kdamball) [reply lib/types.ts:703]: could be `Task`

File: lib/types.ts (hunk 3 above)
Comment (kdamball) [reply lib/types.ts:704]: I think in this case it should be required actually

File: lib/types.ts (hunk 2 above)
Comment (andipaetzold) [reply lib/types.ts:703]: Well then it's still not necessary to have a generic here. And my suggestion would still be fine.

Comment (kdamball) [reply lib/types.ts:703]: Per the conversation, lets keep the generic for consistency 😞

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**